### DISCUSSION TITLE
order of rotation matrix

### DISCUSSION THREAD

**Discussion by TeCai** [GAMES-105/GAMES-105#4] [Q&A] [answered]
关于intrinsic rotation， 刘老师上课的时候说旋转的时候，三个旋转矩阵的组装与extrinsic rotation是反向的。
比如先绕X转a，再绕Y轴转b, 再绕Z转v，对于intrinsic 是$R_x(a)R_y(b)R_z(c)$, 对于extrinsic是$R_z(c)R_y(b)R_x(a)$, 这是为什么呢。并且intrinsic 和 extrinsic 的$R_x(a)$的形式也是一样的吗？

**Comment by KaelMa** [comment]
same question

**Comment by heyuanYao-pku** [comment, marked as answer]
在表示一个旋转矩阵R时，其一般形式为
```math
R = \begin{bmatrix} 
x_1 & y_1 & z_1 \\
x_2 & y_2 & z_2 \\
x_3 & y_3 & z_3
\end{bmatrix} = \begin{bmatrix} x' & y' &z' \end{bmatrix}
```

当你用一个向量a右乘上去时，得到的向量是
```math
a' = Ra = a_1 x'+a_2 y'+a_3 z'
```
故而其一种理解方式为，`原来的x,y,z轴被旋转到了x',y',z'`，而你输入的a是一个会随着基旋转的局部坐标。右乘显式的是旋转后的新的全局坐标。

由此你应该可以理解intrinsic rotation的表示，此时每个Rotation都是把上一次Rotation结果视作新的全局坐标的。故而你求旋转后的结果时需要逆向逐步恢复成真正的全局坐标。

而外旋则是固定了旋转轴在全局坐标。故而每次旋转的矩阵表示都是全局的。所以只需要按旋转顺序乘上去即可。

也可以按照QQ群内的右乘矩阵转自己，左乘矩阵转世界理解。

**Comment by xwc2021** [comment]
v' = R_y v 是1次座標變換
v'’ = R_x R_y v 是2次座標變換
很明顯R_x是父座標系，R_y是子座標系

R_x(a) = R_x(a) |
R_x(a) | 可以看成R_x(a)有個子座標系 I
因為 I是單位矩陣，它的3軸朝向和R_x(a)一樣

R_y(b) 是繞y軸旋轉
R_x(a)R_y(b) = R_x(a)R_y(b) I
R_x(a)R_y(b) I 可以看成
子座標系 I 在父座標系裡繞y軸旋轉
